src/marshmallow/utils.py

```diff
@@ -136,6 +136,11 @@ def __str__(self):
 UTC = utc = UTC()  # UTC is a singleton
 
 
+# https://stackoverflow.com/a/27596917
+def is_aware(datetime):
+    return datetime.tzinfo is not None and datetime.tzinfo.utcoffset(datetime) is not None
```

Comment:
Even after reading the SO post and relevant Python docs, I still don't understand why utcoffset can be `None` when tzinfo is not `None`. But your implementation is correct nonetheless.

>  If d.tzinfo is None, or if d.tzinfo is not None but d.tzinfo.utcoffset(d) returns None, d is naive.

 ¯\\_(ツ)_/¯

Reply:
I'm with you here. I just applied the docs to be on the safe side.

https://docs.python.org/3/library/datetime.html#available-types

> An object of type time or datetime may be naive or aware. A datetime object d is aware if d.tzinfo is not None and d.tzinfo.utcoffset(d) does not return None. If d.tzinfo is None, or if d.tzinfo is not None but d.tzinfo.utcoffset(d) returns None, d is naive. A time object t is aware if t.tzinfo is not None and t.tzinfo.utcoffset(None) does not return None. Otherwise, t is naive.

Reply:
Ideally, Python core would provide an `aware` property to abstract this.